support-frontend/assets/pages/weekly-subscription-checkout/components/weeklyCheckoutFormGifting.jsx

```diff
@@ -230,6 +230,13 @@ function WeeklyCheckoutFormGifting(props: PropTypes) {
               <FieldsetWithError id="startDate" error={firstError('startDate', props.formErrors)} legend="Gift delivery date">
                 {days.map((day) => {
                   const [userDate, machineDate] = [formatUserDate(day), formatMachineDate(day)];
+                  const hideDate = new RegExp('-12-25$').test(machineDate);
```

Comment:
It would be neater to use filter on the collection here before mapping but I'll leave that up to you

Reply:
True, the fix was originally implemented as a semi-urgent request and I felt this was the quickest way to apply the fix and only call `formatMachineDate(day)` once. Now it looks like we should be replacing this logic to [utilise the fulfilment date calculator](https://trello.com/c/bovZtRnp/2788-utilise-fulfilment-date-calculator-to-centralise-logic-for-fulfilment-dates) so this is a temporary fix anyhow.